### ISSUE TITLE
oc_serviceaccount fails to create directory/file for TASK [openshift_manageiq : Create Admin and Image Inspector Service Account]

### ISSUE THREAD

**Issue by shane-dev** [openshift/openshift-ansible#3456]
Trying a containerized based deployment and running into these issue

```
TASK [openshift_manageiq : Create Admin and Image Inspector Service Account] ***
task path: /home/shane/openshift2/openshift-ansible/roles/openshift_manageiq/tasks/main.yaml:21
Using module file /home/shane/openshift2/openshift-ansible/roles/lib_openshift/library/oc_serviceaccount.py
<osmaster01.hidden.domain> ESTABLISH SSH CONNECTION FOR USER: ansible
<osmaster01.hidden.domain> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible -o ConnectTimeout=10 -o ControlPath=/home/shane/.ansible/cp/ansible-ssh-%h-%p-%r osmaster01.hidden.domain '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1487781263.22-275380186412761 `" && echo ansible-tmp-1487781263.22-275380186412761="` echo $HOME/.ansible/tmp/ansible-tmp-1487781263.22-275380186412761 `" ) && sleep 0'"'"''
<osmaster01.hidden.domain> PUT /tmp/tmpMKgO2U TO /home/ansible/.ansible/tmp/ansible-tmp-1487781263.22-275380186412761/oc_serviceaccount.py
<osmaster01.hidden.domain> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible -o ConnectTimeout=10 -o ControlPath=/home/shane/.ansible/cp/ansible-ssh-%h-%p-%r '[osmaster01.hidden.domain]'
<osmaster01.hidden.domain> ESTABLISH SSH CONNECTION FOR USER: ansible
<osmaster01.hidden.domain> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible -o ConnectTimeout=10 -o ControlPath=/home/shane/.ansible/cp/ansible-ssh-%h-%p-%r osmaster01.hidden.domain '/bin/sh -c '"'"'chmod u+x /home/ansible/.ansible/tmp/ansible-tmp-1487781263.22-275380186412761/ /home/ansible/.ansible/tmp/ansible-tmp-1487781263.22-275380186412761/oc_serviceaccount.py && sleep 0'"'"''
<osmaster01.hidden.domain> ESTABLISH SSH CONNECTION FOR USER: ansible
<osmaster01.hidden.domain> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible -o ConnectTimeout=10 -o ControlPath=/home/shane/.ansible/cp/ansible-ssh-%h-%p-%r -tt osmaster01.hidden.domain '/bin/sh -c '"'"'sudo -H -S -n -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-bempzonikkxbwwexwaqgrerlpljrbsxd; /usr/bin/python /home/ansible/.ansible/tmp/ansible-tmp-1487781263.22-275380186412761/oc_serviceaccount.py; rm -rf "/home/ansible/.ansible/tmp/ansible-tmp-1487781263.22-275380186412761/" > /dev/null 2>&1'"'"'"'"'"'"'"'"' && sleep 0'"'"''
failed: [osmaster01.hidden.domain] (item=management-admin) => {
    "failed": true, 
    "invocation": {
        "module_name": "oc_serviceaccount"
    }, 
    "item": "management-admin", 
    "module_stderr": "Shared connection to osmaster01.hidden.domain closed.\r\n", 
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_cLRu6m/ansible_module_oc_serviceaccount.py\", line 1617, in <module>\r\n    main()\r\n  File \"/tmp/ansible_cLRu6m/ansible_module_oc_serviceaccount.py\", line 1610, in main\r\n    rval = OCServiceAccount.run_ansible(module.params, module.check_mode)\r\n  File \"/tmp/ansible_cLRu6m/ansible_module_oc_serviceaccount.py\", line 1516, in run_ansible\r\n    api_rval = oc_sa.get()\r\n  File \"/tmp/ansible_cLRu6m/ansible_module_oc_serviceaccount.py\", line 1449, in get\r\n    result = self._get(self.kind, self.config.name)\r\n  File \"/tmp/ansible_cLRu6m/ansible_module_oc_serviceaccount.py\", line 839, in _get\r\n    rval = self.openshift_cmd(cmd, output=True)\r\n  File \"/tmp/ansible_cLRu6m/ansible_module_oc_serviceaccount.py\", line 965, in openshift_cmd\r\n    returncode, stdout, stderr = self._run(cmds, input_data)\r\n  File \"/tmp/ansible_cLRu6m/ansible_module_oc_serviceaccount.py\", line 936, in _run\r\n    env=curr_env)\r\n  File \"/usr/lib64/python2.7/subprocess.py\", line 711, in __init__\r\n    errread, errwrite)\r\n  File \"/usr/lib64/python2.7/subprocess.py\", line 1327, in _execute_child\r\n    raise child_exception\r\nOSError: [Errno 2] No such file or directory\r\n", 
    "msg": "MODULE FAILURE"
}
Using module file /home/shane/openshift2/openshift-ansible/roles/lib_openshift/library/oc_serviceaccount.py
<osmaster01.hidden.domain> ESTABLISH SSH CONNECTION FOR USER: ansible
<osmaster01.hidden.domain> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible -o ConnectTimeout=10 -o ControlPath=/home/shane/.ansible/cp/ansible-ssh-%h-%p-%r osmaster01.hidden.domain '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1487781263.88-215210194150255 `" && echo ansible-tmp-1487781263.88-215210194150255="` echo $HOME/.ansible/tmp/ansible-tmp-1487781263.88-215210194150255 `" ) && sleep 0'"'"''
<osmaster01.hidden.domain> PUT /tmp/tmpbUcrQV TO /home/ansible/.ansible/tmp/ansible-tmp-1487781263.88-215210194150255/oc_serviceaccount.py
<osmaster01.hidden.domain> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible -o ConnectTimeout=10 -o ControlPath=/home/shane/.ansible/cp/ansible-ssh-%h-%p-%r '[osmaster01.hidden.domain]'
<osmaster01.hidden.domain> ESTABLISH SSH CONNECTION FOR USER: ansible
<osmaster01.hidden.domain> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible -o ConnectTimeout=10 -o ControlPath=/home/shane/.ansible/cp/ansible-ssh-%h-%p-%r osmaster01.hidden.domain '/bin/sh -c '"'"'chmod u+x /home/ansible/.ansible/tmp/ansible-tmp-1487781263.88-215210194150255/ /home/ansible/.ansible/tmp/ansible-tmp-1487781263.88-215210194150255/oc_serviceaccount.py && sleep 0'"'"''
<osmaster01.hidden.domain> ESTABLISH SSH CONNECTION FOR USER: ansible
<osmaster01.hidden.domain> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible -o ConnectTimeout=10 -o ControlPath=/home/shane/.ansible/cp/ansible-ssh-%h-%p-%r -tt osmaster01.hidden.domain '/bin/sh -c '"'"'sudo -H -S -n -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-xtratgucwdogebfhvwyodthikrdthlwq; /usr/bin/python /home/ansible/.ansible/tmp/ansible-tmp-1487781263.88-215210194150255/oc_serviceaccount.py; rm -rf "/home/ansible/.ansible/tmp/ansible-tmp-1487781263.88-215210194150255/" > /dev/null 2>&1'"'"'"'"'"'"'"'"' && sleep 0'"'"''
failed: [osmaster01.hidden.domain] (item=inspector-admin) => {
    "failed": true, 
    "invocation": {
        "module_name": "oc_serviceaccount"
    }, 
    "item": "inspector-admin", 
    "module_stderr": "Shared connection to osmaster01.hidden.domain closed.\r\n", 
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_BH_444/ansible_module_oc_serviceaccount.py\", line 1617, in <module>\r\n    main()\r\n  File \"/tmp/ansible_BH_444/ansible_module_oc_serviceaccount.py\", line 1610, in main\r\n    rval = OCServiceAccount.run_ansible(module.params, module.check_mode)\r\n  File \"/tmp/ansible_BH_444/ansible_module_oc_serviceaccount.py\", line 1516, in run_ansible\r\n    api_rval = oc_sa.get()\r\n  File \"/tmp/ansible_BH_444/ansible_module_oc_serviceaccount.py\", line 1449, in get\r\n    result = self._get(self.kind, self.config.name)\r\n  File \"/tmp/ansible_BH_444/ansible_module_oc_serviceaccount.py\", line 839, in _get\r\n    rval = self.openshift_cmd(cmd, output=True)\r\n  File \"/tmp/ansible_BH_444/ansible_module_oc_serviceaccount.py\", line 965, in openshift_cmd\r\n    returncode, stdout, stderr = self._run(cmds, input_data)\r\n  File \"/tmp/ansible_BH_444/ansible_module_oc_serviceaccount.py\", line 936, in _run\r\n    env=curr_env)\r\n  File \"/usr/lib64/python2.7/subprocess.py\", line 711, in __init__\r\n    errread, errwrite)\r\n  File \"/usr/lib64/python2.7/subprocess.py\", line 1327, in _execute_child\r\n    raise child_exception\r\nOSError: [Errno 2] No such file or directory\r\n", 
    "msg": "MODULE FAILURE"
```

No directories or files are being created on the server from the task , although I do see other  directories are created successfully

Trying to apply 
**openshift_release=v1.5.0-alpha.3**
Due to the known issue around htpasswd_auth using
**ansible 2.2.0.0**
**On branch release-1.5**
Is there any more information I can attach to help with the debugging ?

**Comment by shane-dev** [comment]
So I found the issue. It's because the oc and oadm isn't in the PATH 


In this method 
`def openshift_cmd(self, cmd, oadm=False, output=False, output_type='json', input_data=None):`

It doesn't specify the path and be default , the oc/oadm tools are installed in /usr/local/bin which is outside the 

Temporary fix would be to run on master nodes 
```

sudo  cp /usr/local/bin/oc /usr/bin/
 sudo cp /usr/local/bin/oadm /usr/bin/
```

**Comment by sdodson** [comment]
Fixed in #3416
